src/app/Main/Body/QueryPage/Drawer/Drawer-Layout.component.tsx

```diff
@@ -120,6 +144,13 @@ export function DrawerLayoutComponent(props: DrawerLayoutProps): JSX.Element {
                     onFilterChange={handleChangeFilter}
                     onFilterRemoveAll={handleRemoveAllFilter}
                 />
+                <Button
```

Comment:
As stated a next step could be to move the FilterSettingDialog and button to the DrawerFilterComponent where it belongs to. So your DrawerLayout is just for arranging the two "main" features filterSettings and tableSettings.
A next next step could be to convert the DrawerLayout to a real layout, converting the DrawerFilterComponent to a container (or put one around it) that handles the filter settings and use a filter context in the main container to communicate the filter settings to the FilterSettingsContainer and the ResultsTableContainer. But that should be done after you have a clear picture how the final component structure should look like.

Reply:
Partly done. Commit "Move dialog & make dialog stupid"